pkg/settings/setting_test.go

```diff
@@ -29,3 +31,26 @@ func TestIsRelease(t *testing.T) {
 		a.Equal(value, result, fmt.Sprintf("Expected value [%t] for key [%s]. Got value [%t]", value, key, result))
 	}
 }
+
+// TestSystemDefaultRegistryDefault tests that the default registry is either
+// the value set by the environment variable CATTLE_BASE_REGISTRY or the build
+// time value set through InjectDefaults.
+func TestSystemDefaultRegistryDefault(t *testing.T) {
+	expect := os.Getenv("CATTLE_BASE_REGISTRY")
+	if InjectDefaults != "" {
+		defaults := map[string]string{}
+		if err := json.Unmarshal([]byte(InjectDefaults), &defaults); err != nil {
+			t.Errorf("Unable to parse InjectDefaults: %v", err)
+		}
+
+		if value, ok := settings["system-default-registry"]; ok {
+			expect = value.Get()
+		}
```

Comment:
```suggestion
		if value, ok := defaults["system-default-registry"]; ok {
			expect = value
		}
```
Sorry, I missed this before, but the previous version does not verify that the value from InjectDefaults is being used.

Reply:
Great catch. oops on my part. Fixed.